includes/Checker/Checks/Enqueued_Styles_Scope_Check.php

```diff
@@ -0,0 +1,249 @@
+	 */
+	private $viewable_post_types;
+
+	/**
+	 * List of plugin styles.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $plugin_styles = array();
+
+	/**
+	 * Plugin style counter.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $plugin_style_count = array();
+
+	/**
+	 * Runs this preparation step for the environment and returns a cleanup function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Cleanup function to revert any changes made here.
+	 *
+	 * @throws Exception Thrown when preparation fails.
+	 */
+	public function prepare() {
+		$orig_scripts = isset( $GLOBALS['wp_styles'] ) ? $GLOBALS['wp_styles'] : null;
+
+		// Backup the original values for the global state.
+		$this->backup_globals();
+
+		return function() use ( $orig_scripts ) {
+			if ( is_null( $orig_scripts ) ) {
+				unset( $GLOBALS['wp_styles'] );
+			} else {
+				$GLOBALS['wp_styles'] = $orig_scripts;
+			}
+
+			$this->restore_globals();
+		};
+	}
+
+	/**
+	 * Returns an array of shared preparations for the check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Returns a map of $class_name => $constructor_args pairs. If the class does not
+	 *               need any constructor arguments, it would just be an empty array.
+	 */
+	public function get_shared_preparations() {
+		$demo_posts = array_map(
+			static function( $post_type ) {
+				return array(
+					'post_title'   => "Demo {$post_type} post",
+					'post_content' => 'Test content',
+					'post_type'    => $post_type,
+					'post_status'  => 'publish',
+				);
+			},
+			$this->get_viewable_post_types()
+		);
+
+		return array(
+			Demo_Posts_Creation_Preparation::class => array( $demo_posts ),
+		);
+	}
+
+	/**
+	 * Runs the check on the plugin and amends results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check results to amend and the plugin context.
+	 */
+	public function run( Check_Result $result ) {
+		$urls = $this->get_urls();
+		$this->run_for_urls(
+			$urls,
+			function () use ( $result ) {
+				$this->check_url( $result );
+			}
+		);
+
+		if ( ! empty( $this->plugin_styles ) ) {
+			$url_count = count( $urls );
+			foreach ( $this->plugin_styles as $plugin_style ) {
+				if ( isset( $plugin_style['count'] ) && ( $url_count === $plugin_style['count'] ) ) {
+					$result->add_message(
+						false,
+						__( 'This style is being loaded in all contexts.', 'plugin-check' ),
+						array(
+							'code' => 'EnqueuedStylesScope.StyleLoadedInAllContext',
+							'file' => $plugin_style['path'],
+						)
+					);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Gets the list of URLs to run this check for.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of URL strings (either full URLs or paths).
+	 *
+	 * @throws Exception Thrown when a post type URL cannot be retrieved.
+	 */
+	protected function get_urls() {
+		$urls   = array( home_url( '/' ) );
+		$urls[] = get_search_link();
+		$urls[] = get_author_posts_url( (int) get_the_author_meta( 'ID' ) );
```

Comment:
```suggestion
		$urls   = array( home_url( '/' ) , get_search_link(), get_author_posts_url( 1 ) );
```

Just setup on array like this. Also just hardcode user 1, as this always exists.

Reply:
Thanks for suggestion.